ci/setup_env.sh

```diff
@@ -0,0 +1,114 @@
+    export IS_A_RELEASE="false"
+  else
+    # shellcheck disable=SC2145
+    printf "%s\n" "Tagged build, fetching keys:" "${@}" ""
+    for key in "${my_arr[@]}"; do
+      gpg -v --batch --keyserver hkps://keys.openpgp.org --recv-keys "${key}" ||
+      gpg -v --batch --keyserver hkp://keyserver.ubuntu.com --recv-keys "${key}" ||
+      gpg -v --batch --keyserver hkp://pgp.mit.edu:80 --recv-keys "${key}" ||
+      gpg -v --batch --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys "${key}" ||
+
+      { log bold yellow "WARNING: ${key} can not be found on GPG key servers. Please upload it to at least one of the following GPG key servers:\nhttps://keys.openpgp.org/\nhttps://keyserver.ubuntu.com/\nhttps://pgp.mit.edu/"; export IS_A_RELEASE="false"; }
+    done
+  fi
+}
+
+check_signed_tag() {
+  local tag="${1}"
+
+  if git verify-tag -v "${tag}"; then
+    echo "${tag} is a valid signed tag"
+  else
+    log bold yellow "WARNING: GIT's tag = ${tag} signature is NOT valid. The build is not going to be released ..."
+    export IS_A_RELEASE="false"
+  fi
+}
+
+
+####
+# Main
+####
+log italic green "Production release branch is: ${prod_release_branch}"
+log italic green "Github tag is: ${github_tag}"
+
+# Checking if it is a release build
+if [ -n "${github_tag}" ]; then
+  IS_A_RELEASE="true"
+
+  if [ -z "${MAINTAINERS_KEYS:-}" ]; then
+    log bold yellow "WARNING: MAINTAINERS_KEYS variable is not set. The build is not going to be released ..."
+  fi
+
+  import_gpg_keys "${MAINTAINERS_KEYS}"
+
+  # Checking git tag gpg signature requirement
+  check_signed_tag "${github_tag}"
+
+  # Release test
+  if [ "${IS_A_RELEASE}" = "true" ]; then
+    if (git branch -r --contains "${github_tag}" | grep -xqE ". origin\/${prod_release_branch}$"); then
+      # Checking format of production vs development release version
+      if [[ "${github_tag}" =~ ${prod_release_regex} ]]; then
+        export PROD_RELEASE="true"
+      else
+        log bold yellow "WARNING: GitHub tag: ${github_tag} is in the wrong format for the PRODUCTION release. Expecting the following format: 'd.d.d'. The build is not going to be released ..."
+        export IS_A_RELEASE="false"
+      fi
+    else
+      log bold yellow "WARNING: GitHub tag = ${github_tag} was not created from PRODUCTION branch = ${prod_release_branch}. Trying to build DEVELOPMENT or TEST release"
+      if [[ "${github_tag}" =~ ${dev_release_regex} ]]; then
+        export DEV_RELEASE="true"
+      elif [[ "${github_tag}" =~ ${test_release_regex} ]]; then
+        export TEST_RELEASE="true"
+      else
+        log bold yellow "WARNING: GitHub tag: ${github_tag} is in the wrong format for the DEVELOPMENT or TEST release. Expecting the following format of the release: DEVELOPMENT = 'd.d.d-rc[0-9]' and TEST = 'd.d.d-.*'. The build is not going to be released ..."
+        export IS_A_RELEASE="false"
+      fi
+    fi
+  fi
```

Comment:
Please rewrite this as below:
```
  # Release test
  if [ "${IS_A_RELEASE}" = "true" ]; then
    if (git branch -r --contains "${github_tag}" | grep -xqE ". origin\/${prod_release_branch}$"); then
      # Checking format of production vs development release version
      if [[ "${github_tag}" =~ ${prod_release_regex} ]]; then
        export PROD_RELEASE="true"
      elif [[ "${github_tag}" =~ ${dev_release_regex} ]]; then
        export DEV_RELEASE="true"
      elif [[ "${github_tag}" =~ ${test_release_regex} ]]; then
        export TEST_RELEASE="true"
      else
        log bold yellow "WARNING: GitHub tag: ${github_tag} is in the wrong format for any type of release (PRODUCTION, DEVELOPMENT, TEST). The build is not going to be released ..."
        export IS_A_RELEASE="false"
      fi
    else
      if [[ "${github_tag}" =~ ${prod_release_regex} ]]; then
        log bold yellow "WARNING: GitHub tag = ${github_tag} is in the PRODUCTION format but was not created from PRODUCTION branch = ${prod_release_branch}."
        export IS_A_RELEASE="false"
      elif [[ "${github_tag}" =~ ${dev_release_regex} ]]; then
        export DEV_RELEASE="true"
      elif [[ "${github_tag}" =~ ${test_release_regex} ]]; then
        export TEST_RELEASE="true"
      else
        log bold yellow "WARNING: GitHub tag: ${github_tag} is in the wrong format for the DEVELOPMENT or TEST release. The build is not going to be released ..."
        export IS_A_RELEASE="false"
      fi
    fi
  fi
```
Apart from logs adaptation, the main point here is that we could want (and actually almost always do this way) to create tags on `main`, also for dev/test versions.

Reply:
Spoke with @lander86 about it and seems like we should not be releasing anything from main rather than PROD and should use development branch or smth like this for this kind of scenario